packages/grid/x-data-grid/src/models/colDef/gridColType.ts

```diff
@@ -7,4 +7,18 @@ export type GridNativeColTypes =
   | 'singleSelect'
   | 'actions';
 
-export type GridColType = GridNativeColTypes | string;
+/**
+ * Workaround for Typescript losing type information when a string literal is opened up to any string
+ * 
+ * Can be removed if this issue is ever resolved:
+ * https://github.com/Microsoft/TypeScript/issues/29729
+ */
+type LiteralAnyString = string & Record<never, never>
+
+/**
+ * Immitate typescript's template literal syntax to document in autocomplete that the string type is open for extension.
+ */
+// eslint-disable-next-line no-template-curly-in-string
+type LiteralAnyStringDocPrompt = '${string}'
```

Comment:
So this is a bit of an experiment. 

* When seeing autocomplete in typescript, it can appear to developers that you're restricted from using any value which deviates from the listed values
* Typescript has established syntax in template literals for literals with some variable/free-text element
* Typescript does not have an established approach for documenting when literals and template-literals permit "any string" as a value, it simply disabled autocomplete without workarounds like implemented here.

So my idea is to list the options with a template literal shown to help communicate that other values are permitted:

<img width="790" alt="image" src="https://user-images.githubusercontent.com/8896153/207326382-d245d56b-ec30-4661-bb23-9be1285201d9.png">